contrib/babelfishpg_tds/src/backend/tds/err_handler.c

```diff
@@ -195,8 +195,12 @@ get_tsql_error_details(ErrorData *edata,
 
 		/* Possible infinite loop of errors. Do not touch it further. */
 		if (!error_stack_full())
```

Comment:
how are we testing these? do we have test cases which exercise each and every code path where we are adding HOLD/RESUME?

Reply:
We are doing this in the error path so I am sure that we have test cases for it. Now, how effective these HOLD/RESUMEs are always dependent on timing b/w leader node and worker node.